server/fleet/activities.go

```diff
@@ -4,48 +4,416 @@ import (
+	return "deleted_saved_query"
+}
+
+func (a ActivityTypeDeletedSavedQuery) Documentation() (string, string, string) {
+	return `Generated when deleting a saved query.`,
+		`This activity contains the following fields:
+- "query_name": the name of the query being deleted.`, `{
+	"query_name": "foo"
+}`
+}
+
+type ActivityTypeDeletedMultipleSavedQuery struct {
+	IDs []uint `json:"query_ids"`
+}
+
+func (a ActivityTypeDeletedMultipleSavedQuery) ActivityName() string {
+	return "deleted_multiple_saved_query"
+}
+
+func (a ActivityTypeDeletedMultipleSavedQuery) Documentation() (string, string, string) {
+	return `Generated when deleting multiple saved queries.`,
+		`This activity contains the following fields:
+- "query_ids": list of IDs of the deleted saved queries.`, `{
+	"query_ids": [1, 42, 100]
+}`
+}
+
+type ActivityTypeAppliedSpecSavedQuery struct {
+	Specs []*QuerySpec `json:"specs"`
+}
+
+func (a ActivityTypeAppliedSpecSavedQuery) ActivityName() string {
+	return "applied_spec_saved_query"
+}
+
+func (a ActivityTypeAppliedSpecSavedQuery) Documentation() (string, string, string) {
+	return `Generated when applying a query spec.`,
+		`This activity contains a field "specs" where each item is a query spec with the following fields:
+- "name": Name of the query.
+- "description": Description of the query.
+- "query": SQL query.`, `{
+	"specs": [
+		{
+			"name":"Get OpenSSL versions",
+			"query":"SELECT name AS name, version AS version, 'deb_packages' AS source FROM [...]",
+			"description":"Retrieves the OpenSSL version."
+		}
+	]
+}`
+}
+
+type ActivityTypeCreatedTeam struct {
+	ID   uint   `json:"team_id"`
+	Name string `json:"team_name"`
+}
+
+func (a ActivityTypeCreatedTeam) ActivityName() string {
+	return "created_team"
+}
+
+func (a ActivityTypeCreatedTeam) Documentation() (string, string, string) {
+	return `Generated when creating teams.`,
+		`This activity contains the following fields:
+- "team_id": unique ID of the created team.
+- "team_name": the name of the created team.`, `{
+	"team_id": 123, 
+	"team_name": "foo"
+}`
+}
+
+type ActivityTypeDeletedTeam struct {
+	ID   uint   `json:"team_id"`
+	Name string `json:"team_name"`
+}
+
+func (a ActivityTypeDeletedTeam) ActivityName() string {
+	return "deleted_team"
+}
+
+func (a ActivityTypeDeletedTeam) Documentation() (string, string, string) {
+	return `Generated when deleting teams.`,
+		`This activity contains the following fields:
+- "team_id": unique ID of the deleted team.
+- "team_name": the name of the deleted team.`, `{
+	"team_id": 123, 
+	"team_name": "foo"
+}`
+}
+
+type TeamActivityDetail struct {
+	ID   uint   `json:"id"`
+	Name string `json:"name"`
+}
+
+type ActivityTypeAppliedSpecTeam struct {
+	Teams []TeamActivityDetail `json:"teams"`
+}
+
+func (a ActivityTypeAppliedSpecTeam) ActivityName() string {
+	return "applied_spec_team"
+}
+
+func (a ActivityTypeAppliedSpecTeam) Documentation() (string, string, string) {
```

Comment:
HUH! Wonder if we can use something like this for API docs.

Reply:
I don't like it, but it really forces the developer to write the docs (we are lazy...)